routes/mylibrary.js

```diff
@@ -97,52 +100,99 @@ function addUserBook(userId, bookId, conditionId, callback)
 // @desc    Get current users mylibrary
 router.get('/', common.isAuthenticated, (req, res, next) => {
     const userId = req.session.userId;
-    var payload = {
-        title: 'My Library',
+    var payload = { 
+        title: 'My Library', 
         library: [],
-        titles: [],
-        conditions: [],
         avail: 0,
         rcvd: 0
     };
-    var selectAllBooks = `SELECT UserBooks.id AS userBookId, Books.id AS bookId, available AS swap, listingDate AS date, Books.title AS title, Books.imgUrl AS imgUrl
+    var selectAllBooks = `SELECT UserBooks.id AS userBookId, available AS swap, listingDate AS date, description, Books.author As author, Books.title AS title, Books.imgUrl AS imgUrl
                         FROM UserBooks
                         INNER JOIN Books ON Books.id = UserBooks.bookId
-                        WHERE UserBooks.userId = ?`
-    var selectAllConditions = 'SELECT id, description FROM Conditions ORDER BY id'
-    var selectAllTitles = 'SELECT isbn13, title FROM Books ORDER BY title'
+                        INNER JOIN Conditions on Conditions.id = UserBooks.conditionId
+                        WHERE UserBooks.userId = ? AND UserBooks.available = 1
+                        ORDER BY title ASC`;
+    var selectRCVDBooks = `SELECT UserBooks.id AS userBookId, date, description, available AS swap, Books.author As author, Books.title AS title, Books.imgUrl AS imgUrl
+                        FROM Books
+                        INNER JOIN UserBooks ON UserBooks.bookId = Books.id
+                        INNER JOIN Transactions ON Transactions.userBookId = UserBooks.id 
+                        INNER JOIN TransactionStatusDates ON TransactionStatusDates.transactionId = Transactions.id
+                        INNER JOIN Conditions on Conditions.id = UserBooks.conditionId
+                        WHERE Transactions.requestorId = ? AND Transactions.statusId = 8 AND TransactionStatusDates.statusId = 4`;
```

Comment:
Per recent PR's, it was noted that the closed logic had to change a little bit. There is no status 8 anymore and it's a flag instead. Although for this we don't care about that.

You should be able to remove the Inner join to transaction status dates completely as it's irrelevant that's just a history of the statuses. We don't care if it's closed or not because once the user receives the book it is received regardless of whether the points have been 'confirmed'. This query should only need to look at Transactions where statusId = 4 which will be the last status it will go into once received